calendar-bundle/src/EventListener/BreadcrumbListener.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CalendarBundle\EventListener;
+
+use Contao\CalendarEventsModel;
+use Contao\CalendarModel;
+use Contao\Config;
+use Contao\CoreBundle\Framework\Adapter;
+use Contao\CoreBundle\Framework\ContaoFrameworkInterface;
+use Contao\Events;
+use Contao\Input;
+use Contao\PageModel;
+use Contao\StringUtil;
+
+class BreadcrumbListener
+{
+    /**
+     * @var ContaoFrameworkInterface
+     */
+    private $framework;
+
+    public function __construct(ContaoFrameworkInterface $framework)
+    {
+        $this->framework = $framework;
+    }
+
+    public function onGenerateBreadcrumb(array $items): array
+    {
+        $eventAlias = $this->getEventAlias();
+        if (!$eventAlias) {
+            return $items;
+        }
+
+        $calendar = $this->getCalenderModel();
+        if (!$calendar) {
+            return $items;
+        }
+
+        $eventModel = $this->getEventModel($eventAlias, $calendar);
+        if (!$eventModel) {
+            return $items;
+        }
+
+        if ($GLOBALS['objPage']->requireItem) {
+            return $this->overrideActiveBreadcrumbItem($items, $eventModel);
+        }
+
+        return $this->addBreadcrumbItem($items, $eventModel);
+    }
+
+    private function getEventAlias(): ?string
+    {
+        if (!isset($GLOBALS['objPage'])) {
```

Comment:
When does this case occur?

Reply:
For instance, if the breadcrumb module is loaded via an esi request which hasn't setup the page context. In general it's a global state outside of the control of this listener. In the sense of defensive programming, I check it here.

Reply:
@Toflar Is this possible?